server.js

```diff
@@ -0,0 +1,182 @@
+};
+
+//Middlewares
+
+function productValidator(req, res, next){
+    if(req.body.id && req.body.nombre && req.body.precio && req.body.url){
+        let match = false;
+        productos.forEach(e => {
+            if(e.id == req.body.id){
+                match = true
+            }
+        });      
+        if(match) {
+            res.status(409).json('El id ya existe');
+        } else {
+            req.body.id = parseInt(req.body.id);
+            req.body.precio = parseInt(req.body.precio);
+            req.body.isFavorite === 'true' ? req.body.isFavorite = true : req.body.isFavorite = false;
+            next();
+        }       
+    } else {
+        res.status(400).json('Faltan parametros');
+    }  
+};
+
+function userPassValidator(mail, contrasena) {
+    const [filterUser] = usuarios.filter(element => element.mail === mail && element.contrasena === contrasena);
+    if (!filterUser) {
+        return false;
+    }
+    return filterUser;
+};
+
+function userAuthenticaton(req, res, next) {
+    try {
+        const token = req.headers.authorization.split(' ')[1];
+        const verifyToken = jwt.verify(token, claveSegura);
+        if (verifyToken) {
+            req.usuario = verifyToken;
+            return next();
+        }
+    } catch (err) {
+        res.json({ error: 'Error al validar el usuario'});
+    }
+};
+
+function newUserVerify(mail) {
+    const [userInvalid] = (usuarios.filter(element => element.mail === mail));
+    if(!userInvalid) {
+        return false;
+    }
+    return userInvalid   
+};
+    
+
+//Paths
+
+server.post('/login', allowCors, (req, res) => {
+    const { mail, contrasena } = req.body;
+    const validated = userPassValidator(mail, contrasena);
+    if (!validated) {
+        res.json({error: 'No existe el usuario o la contraseña es incorrecta'});
+        return
+    }
+
+    const token = jwt.sign({ mail }, claveSegura);
+
+    res.json({ token })
+
+});
+
+server.get('/auth', [userAuthenticaton, allowCors], (req, res) => {
+    res.send(`Esta es una página autenticada. Hola ${req.usuario.mail} !`);
+})
+
+server.get('/productos', allowCors, (req, res) => {
+    res.json(productos);
+});
+
+server.get('/productos/:idproducto', allowCors, (req, res) => {
+    const idproducto = req.params.idproducto;
+    if(idproducto > productos.length || idproducto == 0) {
+        return res.status(404).send('Artículo no encontrado');
+    }
+    res.json(productos[idproducto-1]);
+});
+
+server.post('/productos', [productValidator, allowCors], (req, res) => {
+    productos.push(req.body);
+    console.log('El producto ha sido agregado con exito');
+    console.log(productos);
+    res.json(req.body);
+});
+
+server.get('/productos-favoritos', allowCors, (req, res) => {
+    const favorites = productos.filter(elem => elem.isFavorite == true);
+    console.log(favorites);
+    res.json(favorites);
+    
+});
+
+server.post('/register', allowCors, (req, res) => {
+    const { mail } = req.body
```

Comment:
if it is only one variable you can use const mail = req.body.mail; but is okay on either ways